execution/evm/vm.go

```diff
@@ -1145,16 +1150,17 @@ func (vm *VM) call(caller acm.Account, callee acm.MutableAccount, code, input []
 			if errAdd != nil {
 				return nil, firstErr(err, errAdd)
 			}
-			vm.stateWriter.UpdateAccount(receiver)
-			vm.stateWriter.RemoveAccount(callee.Address())
+			callState.UpdateAccount(receiver)
+			callState.RemoveAccount(callee.Address())
 			vm.Debugf(" => (%X) %v\n", addr[:4], callee.Balance())
 			fallthrough
 
 		case STOP: // 0x00
 			return nil, nil
 
-		case STATICCALL, CREATE2:
+		case STATICCALL:
```

Comment:
This PR is not changing anything wrt to CREATE2

Reply:
I made the changes including Silas' request for a comment above.

Reply:
Some tests are now not passing